### ISSUE TITLE
`Invoke-VCFCommand` cmdlet assumes Windows and fails on MacOS and Linux

### ISSUE THREAD

**Issue by nathanthaler** [vmware/powershell-module-for-vmware-cloud-foundation#260]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### PowerVCF

2.5.0.0

### VMware Cloud Foundation

5.1

### PowerShell

7.4.1

### Guest Operating System

MacOS 14.2.1 (x86-64)

### Environment Details

```powershell
$▶ /usr/bin/ssh -V
OpenSSH_9.4p1, LibreSSL 3.3.6

$▶ uname -a
Darwin {{redacted}} 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:10 PST 2023; root:xnu-10002.61.3~2/RELEASE_X86_64 x86_64

$▶ pwsh -V
PowerShell 7.4.1
```

### Description

```powershell
>         Invoke-VCFCommand -vcfpassword VMware1! -rootPassword VMware1! -sosOption general-health
Invoke-VCFCommand: PowerShell Module Posh-SSH staus is: Module Posh-SSH was missing, installing now please wait... Importing module Posh-SSH, please wait... INSTALLIMPORT_FAILED. Posh-SSH is required to run this cmdlet. Please install the module and try again.
```

### Expected Behavior

Install `Posh-SSH` and utilize it if and only if the GuestOS is Windows.  

For MacOS and Linux, running Powershell Core, should be able to use the system SSH client. 

### Actual Behavior

Command fails with:

```powershell
Invoke-VCFCommand: PowerShell Module Posh-SSH staus is: Module Posh-SSH was missing, installing now please wait... Importing module Posh-SSH, please wait... INSTALLIMPORT_FAILED. Posh-SSH is required to run this cmdlet. Please install the module and try again.
```

### Steps to Reproduce

Run the command on a non-Windows platform.  

Ex:

``powershell
 Invoke-VCFCommand -vcfpassword VMware1! -rootPassword VMware1! -sosOption general-health
``

**Comment by nathanthaler** [comment]
This seems to be specific to running get-sos.  Can this be removed based on the availability of the health API?

**Comment by tenthirtyam** [comment]
After taking a review of this function and it's intended purpose, I think we should consider removing this functionality in favor of the SoS functions that are included for 4.4 and later.

- `Get-VCFHealthSummaryTask`
- `Request-VCFHealthSummaryBundle`
- `Start-VCFHealthSummary`

These began to be used by `VMware.CloudFoundation.Reporting` after the above additions were made.

That said, there may be other internal tools that we'll need to validate that those are not using the function.

@LifeOfBrianOC and @GaryJBlake for comment.
